src/tests/test_exports.py

```diff
@@ -0,0 +1,1101 @@
+        """Test export handles database with no released genomes gracefully."""
+        metadata_uri = test_dbs['ensembl_genome_metadata'].dbc.url
+        exporter = FTPMetadataExporter(metadata_uri)
+        metadata = exporter.export_to_json()
+        assert metadata is not None
+        assert 'last_updated' in metadata
+        assert 'species' in metadata
+        assert isinstance(metadata['species'], dict)
+
+    def test_json_file_is_valid_json(self, test_dbs, tmp_path):
+        """Test that exported JSON file can be read back and is valid."""
+        metadata_uri = test_dbs['ensembl_genome_metadata'].dbc.url
+        exporter = FTPMetadataExporter(metadata_uri)
+        output_file = tmp_path / "test_output.json"
+        exporter.export_to_json(str(output_file))
+
+        with open(output_file, 'r') as f:
+            data = json.load(f)
+        assert 'last_updated' in data
+        assert 'species' in data
+        from datetime import datetime
+        try:
+            datetime.fromisoformat(data['last_updated'])
+        except ValueError:
+            pytest.fail("last_updated is not in valid ISO format")
+
+
+@pytest.mark.parametrize("test_dbs", [[{'src': Path(__file__).parent / "databases/ensembl_genome_metadata"},
+                                       {'src': Path(__file__).parent / "databases/ncbi_taxonomy"},
+                                       ]], indirect=True)
+class TestChangelogGenerator:
+    """Test suite for ChangelogGenerator class."""
+
+    def test_init_valid_parameters(self, test_dbs):
+        """Test ChangelogGenerator initialization with valid parameters."""
+        metadata_uri = test_dbs['ensembl_genome_metadata'].dbc.url
+        generator = ChangelogGenerator(
+            metadata_uri=metadata_uri,
+            release_label="2024-01"
+        )
+        assert generator.metadata_db is not None
+        assert generator.release_label == "2024-01"
+        assert generator.output_path is None
+
+    def test_init_with_output_path(self, test_dbs):
+        """Test ChangelogGenerator initialization with custom output path."""
+        metadata_uri = test_dbs['ensembl_genome_metadata'].dbc.url
+        output_path = "/tmp/test_changelog.csv"
+
+        generator = ChangelogGenerator(
+            metadata_uri=metadata_uri,
+            release_label="2024-01",
+            output_path=output_path
+        )
+        assert generator.metadata_db is not None
+        assert generator.release_label == "2024-01"
+        assert generator.output_path == output_path
+
+    def test_init_invalid_metadata_uri_empty(self, test_dbs):
+        """Test initialization fails with empty metadata URI."""
+        with pytest.raises(ValueError) as excinfo:
+            ChangelogGenerator(
+                metadata_uri="",
+                release_label="2024-01"
+            )
+        assert "metadata_uri must be a non-empty string" in str(excinfo.value)
```

Comment:
```suggestion
        with pytest.raises(ValueError, match="metadata_uri must be a non-empty string"):
            ChangelogGenerator(
                metadata_uri="",
                release_label="2024-01"
            )
```
And similarly below.